app/scripts/directives/mobileServiceInstanceRow.js

```diff
@@ -0,0 +1,125 @@
+    templateUrl: 'views/_mobile-service-instance-row.html'
+  });
+
+  function MobileServiceInstanceRow($filter,
+                              $routeParams,
+                              $scope,
+                              APIService,
+                              AuthorizationService,
+                              BindingService,
+                              DataService,
+                              ListRowUtils) {
+    var row = this;
+    var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
+    row.serviceBindingsVersion = APIService.getPreferredVersion('servicebindings');
+    row.state = row.state || {};
+    var annotationSpace = "org.aerogear.binding." + this.apiObject.metadata.name;
+    var extendedAnnotationSpace = "org.aerogear.binding-ext." + this.apiObject.metadata.name;
+    var watches = [];
+    var bindingsWatch = false;
+    row.bindingsLimit = _.get(row.serviceClass, "spec.externalMetadata.bindingsLimit", 1);
+    _.extend(row, ListRowUtils.ui);
+
+    $scope.$on('$destroy', function(){
+        DataService.unwatchAll(watches);
+        DataService.unwatch(bindingsWatch);
+    });
+
+    watches.push(DataService.watchObject(APIService.getPreferredVersion('mobileclients'), $routeParams.mobileclient, {namespace: $routeParams.project}, function(mobileClient) {
+        row.mobileClient = mobileClient;
+
+        row.annotations = _.filter(row.mobileClient.metadata.annotations, function(annotation, name){
+          return name.split("/")[0] === annotationSpace;
+        })
+        .map(JSON.parse)
+        .sort(function(a, b) {
+          if (a.label < b.label) {
+            return -1;
+          }
+          if (a.label > b.label) {
+            return 1;
+          }
+          return 0;
+        });
+
+        row.serviceType = _.get(row, "serviceClass.spec.externalMetadata.serviceName"); // safe get
+
+        row.extendedAnnotations = _(row.mobileClient.metadata.annotations)
+          .pickBy(function(annotation, name){
+            return name.split("/")[0] === extendedAnnotationSpace;
+          })
+          .transform(function(result, strVal, name) {
+            var key = name.split("/")[1];
+            result[key] = JSON.parse(strVal);
+          })
+          .value();
+
+        row.servicePlanPromise = DataService.list(APIService.getPreferredVersion('clusterserviceplans'), {}, function(servicePlans) {
+          var servicePlanData = servicePlans.by('metadata.name');
+          row.providerServicePlan = _.find(servicePlanData, function(plan) {
+            return row.serviceClass.metadata.name === plan.spec.clusterServiceClassRef.name;
+          });
+          var kind = row.mobileClient.kind.toLowerCase();
+          var bindDataPath = 'providerServicePlan.spec.externalMetadata.' + kind + '_bind_parameters_data';
+          var bindData = _.get(row, bindDataPath, []);
+          row.bindData = _(bindData).map(JSON.parse).value();
+        });
+
+        if(!bindingsWatch){
```

Comment:
`_.once()` to eliminate `bindingsWatch ` check.

Reply:
This watch is removed in favour of a get on the mobile client instead of `watchObject`